src/emoji-row.jsx

```diff
@@ -0,0 +1,43 @@
+import React, {Component, PropTypes} from 'react';
+import shallowCompare from 'react-addons-shallow-compare';
+import Emoji from './emoji';
+
+export default class EmojiRow extends Component {
+  static propTypes = {
+    emojis: PropTypes.array.isRequired,
+    onChange: PropTypes.func.isRequired,
+    style: PropTypes.object.isRequired,
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return shallowCompare(this, nextProps, nextState);
+  }
+
+  render() {
+    const {emojis, onChange, style} = this.props;
+
+    return (
+      <div className="emoji-row" style={style}>
+        {emojis.map(emoji => {
+          return (
+            <Emoji
+              {...emoji}
+              ariaLabel={emoji.name}
+              role="option"
+              key={emoji.unicode}
+              onClick={function(e) {
+                onChange(emoji);
```

Comment:
How would you prefer to do this? The change handlers need to be bound to to the emoji object being selected, so I believe we'll at least need to allocate at least one function per emoji.

One possibility -
The onClick and onKeyUp handlers are both intended to select the emoji, so we could define the onClick and onKeyUp handlers in the Emoji component instead, and the component rendering the Emoji component could pass a single onSelect prop instead. That should save one function allocation per emoji, since the onClick and onKeyUp functions would be shared on the Emoji prototype. It would also have the benefit of consolidating the keyup key filtering logic, because the Emoji component is rendered by two different components.